src/Relation/Nullary/Construct/Add/Point.agda

```diff
@@ -8,8 +8,84 @@
 
 module Relation.Nullary.Construct.Add.Point where
 
+open import Algebra.Bundles
+open import Algebra.Core using (Op₂)
+import Algebra.Definitions as Definitions
+import Algebra.Structures as Structures
+open import Level using (Level)
 open import Data.Maybe.Base public
   using () renaming (Maybe to Pointed; nothing to ∙; just to [_])
 
 open import Data.Maybe.Properties public
   using (≡-dec) renaming (just-injective to []-injective)
+open import Data.Product using (_,_)
+open import Relation.Binary.PropositionalEquality
+
+private
+  variable
+    a : Level
+
+module _ {A : Set a} {op : Op₂ A} where
+  open Definitions {A = Pointed A} _≡_
+  private
+    module Str = Structures {A = A} _≡_
+  module _ {semigroup : Str.IsSemigroup op} where
```

Comment:
Hmm not every proof in this file requires the underlying op to be a semigroup right? Most of the basic definitions simply require reflexivity of the equality. Could you parameterise the proofs by just the locally required properties in instead?

Reply:
Yes! Roughly, what is the best way to do that? Shall I just add the semigroup assumption as an argument to associativity or something else?

Reply:
I would just add associativity as an assumption right? Unless I'm misunderstanding something we don't need the full semigroup?

Reply:
Ah, got it.